src/nc_server.c

```diff
@@ -453,6 +456,14 @@ server_close(struct context *ctx, struct conn *conn)
 
     conn->unref(conn);
 
+    if (conn->ssl) {
```

Comment:
if you wanted, you could be all like: `nc_teardown_ssl_if_necessary(&conn->ssl)` and then all like:
```
void nc_teardown_ssl_if_necessary(struct conn **con)
{
    if (*conn->ssl == NULL) {
        return;
    }

    if (nc_teardown_ssl(*conn->ssl) != NC_OK) {
        log_error("failed to teardown ssl s %d", *conn->sd);
    }

    *conn->ssl = NULL;
}
```

Reply:
👍 This is a pretty long function, I like that idea.